pkg/kubelet/dockertools/labels.go

```diff
@@ -30,52 +32,69 @@ import (
-	kubernetesPodUID            = "io.kubernetes.pod.uid"
+	kubernetesPodUIDLabel       = "io.kubernetes.pod.uid"
 
-	kubernetesContainerRestartCountLabel      = "io.kubernetes.container.restartCount"
-	kubernetesContainerTerminationMessagePath = "io.kubernetes.container.terminationMessagePath"
+	kubernetesContainerNameLabel                   = "io.kubernetes.container.name"
+	kubernetesContainerHashLabel                   = "io.kubernetes.container.hash"
+	kubernetesContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
+	kubernetesContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
 
 	kubernetesPodLabel                    = "io.kubernetes.pod.data"
 	kubernetesTerminationGracePeriodLabel = "io.kubernetes.pod.terminationGracePeriod"
 	kubernetesContainerLabel              = "io.kubernetes.container.name"
 )
 
+// Container information which has been labelled on each docker container
+type labelledContainerInfo struct {
+	PodName                string
+	PodNamespace           string
+	PodUID                 types.UID
+	Name                   string
+	Hash                   string
+	RestartCount           int
+	TerminationMessagePath string
+}
+
 func newLabels(container *api.Container, pod *api.Pod, restartCount int) map[string]string {
 	// TODO (random-liu) Move more label initialization here
 	labels := map[string]string{}
 	labels[kubernetesPodNameLabel] = pod.Name
 	labels[kubernetesPodNamespaceLabel] = pod.Namespace
-	labels[kubernetesPodUID] = string(pod.UID)
+	labels[kubernetesPodUIDLabel] = string(pod.UID)
 
+	labels[kubernetesContainerNameLabel] = container.Name
+	labels[kubernetesContainerHashLabel] = strconv.FormatUint(kubecontainer.HashContainer(container), 16)
 	labels[kubernetesContainerRestartCountLabel] = strconv.Itoa(restartCount)
-	labels[kubernetesContainerTerminationMessagePath] = container.TerminationMessagePath
+	labels[kubernetesContainerTerminationMessagePathLabel] = container.TerminationMessagePath
 
 	return labels
 }
 
-func getRestartCountFromLabel(labels map[string]string) (restartCount int, err error) {
-	if restartCountString, found := labels[kubernetesContainerRestartCountLabel]; found {
-		restartCount, err = strconv.Atoi(restartCountString)
-		if err != nil {
-			// This really should not happen. Just set restartCount to 0 to handle this abnormal case
-			restartCount = 0
-		}
-	} else {
-		// Get restartCount from docker label. If there is no restart count label in a container,
-		// it should be an old container or an invalid container, we just set restart count to 0.
-		// Do not report error, because there should be many old containers without this label now
-		glog.V(3).Infof("Container doesn't have label %s, it may be an old or invalid container", kubernetesContainerRestartCountLabel)
+func getContainerInfoFromLabel(labels map[string]string) (*labelledContainerInfo, error) {
+	var err error
+	containerInfo := labelledContainerInfo{}
```

Comment:
Just a minor suggestion:

It may look cleaner to do
```
containerInfo := labelledContainerInfo{
  PodName: getStringValueFromLabel(...),
  PodNamespace: getStringValueFromLabel(...),
  ...
}

Another option is to use a shorter variable (e.g., `c') name since this is a small function.